nuxeo-ai-core/src/main/java/org/nuxeo/ai/configuration/ThresholdComponent.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Gethin James
+ */
+package org.nuxeo.ai.configuration;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.nuxeo.ai.configuration.ThresholdConfiguratorDescriptor.Configuration;
+import org.nuxeo.ai.configuration.ThresholdConfiguratorDescriptor.Threshold;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.runtime.model.ComponentInstance;
+import org.nuxeo.runtime.model.DefaultComponent;
+
+/**
+ * Implementation of the ThresholdService
+ */
+public class ThresholdComponent extends DefaultComponent implements ThresholdService {
+
+    public static final String THRESHOLD_CONFIGURATION_XP = "thresholdConfiguration";
+
+    protected float globalThreshold = 0.1f;
+
+    protected final Map<String, Map<String, Float>> thresholds = new HashMap<>();
+
+    @Override
+    public void registerContribution(Object contribution, String xp, ComponentInstance component) {
```

Comment:
When you are using the `Descriptor` interface this should be `start` rather than `registerContribution`.

Reply:
I don't see a point to keep a list of descriptors that have no use.

Reply:
I will forward the email to you from Pierre.

Reply:
which describes how registring can be done simpler. but at the same time there still will be a list of descriptors that I don't use. I just need the map of type->prop->value
so I'm constructing one